### ISSUE TITLE
tests: 99999 should be RSIZE_MAX_STR+1

### ISSUE THREAD

**Issue by ddurham2** [rurban/safeclib#88]
On this line of code: 

https://github.com/rurban/safeclib/blob/a7da29dadadde04feec174f595b59d67a64f3956/tests/test_strtouppercase_s.c#L33

99999 should be RSIZE_MAX_STR+1

Similarly, RSIZE_MAX_MEM+1 is used in the memcpy32 / memcpy16 / memcmp32 / memcmp16 / ... etc tests.  Shouldn't those be RSIZE_MAX_MEM/4 + 1 and RSIZE_MAX_MEM/2 + 1 respectively?   These tests failed when I had RSIZE_MAX_MEM set to 1GB.    

But I think there is a deeper issue, inside of memcpy32 (possible others). When RSIZE_MAX_MEM was 1GB, it _did not_ fail with overflow which means there is an internal overflow when computing the limit.  I would expect it to return an error code since you should only be able to pass /4 of the underlying limit.

Cheers.

**Comment by rurban** [comment]
BTW leaving RSIZE_MAX_MEM at 1GB is certainly not safe. It allows DOS attacks. Allocating and using such big buffers need an overlong time even on my fastest machine.
You need to use unsafe functions for that. 

eg. 
```
$ time perl5.28.2 -e'$a[10000000000]=0'
Out of memory!

real	0m13.891s
user	0m0.005s
sys	0m0.021s
```

**Comment by rurban** [comment]
All issues fixed with c8bed9d3f40133a6e83f4c87e273b09913edd962

test_strtouppercase_s.c uses RSIZE_MAX_STR + 1 for the dest overflow check.
test_memcpy16_s uses RSIZE_MAX_MEM + 1 for the dest overflow and RSIZE_MAX_MEM16 + 1 fr the src overflow checks. ditto all others.

Thanks
